benchmarks/views/user.py

```diff
@@ -226,87 +224,117 @@ def is_submission_original(file, submitter: User) -> Tuple[bool, Union[None, Lis
+    and their instances, and verifies that each instance contains specific files.
+
+    :param file: Path to the zip file.
+    :return: Tuple containing a boolean for success and an error message if any.
+    """
     with zipfile.ZipFile(file, mode="r") as archive:
         namelist = archive.infolist()
         root = namelist[0]
         has_plugin, submitted_plugins = plugins_exist(namelist)
-        if not has_plugin:
-            return False, f"\nPlease make sure your {root.filename} folder contains at least one of the " \
-                          f"following folders:" \
-                          "[metrics, data, benchmarks, models]"
-        instances = []
-        files = []
-        for plugin in submitted_plugins:
-            has_instance, submitted_instances = plugin_has_instances(namelist, plugin)
-            instances.append(submitted_instances)
-        plugin_instance_dict = dict(zip(submitted_plugins, instances))
-
-        # make sure there is at least one plugin that is not empty:
-        if all(x == [] for x in plugin_instance_dict.values()):
-
-            if len(list(plugin_instance_dict.keys())) == 1:
-                return False, f"\nYour {list(plugin_instance_dict.keys())} folder is empty."
+        if not has_plugin:  # checks for at least one plugin
+            return False, (f"\nPlease make sure your {root.filename} folder contains at least one of the "
+                           "following valid plugin folders: [metrics, data, benchmarks, models].")
+
+        plugin_instance_dict = {plugin: plugin_has_instances(namelist, plugin)[1] for plugin in submitted_plugins}
+        if all(not instances for instances in plugin_instance_dict.values()):
+            folder_names = list(plugin_instance_dict.keys())
+            if len(folder_names) == 1:
+                return False, f"\nYour {folder_names[0]} folder is empty."
             else:
-                return False, f"\nYour {list(plugin_instance_dict.keys())} folders are empty."
+                return False, f"\nYour {', '.join(folder_names)} folders are empty."
 
-        for instance in plugin_instance_dict.values():
-            if len(instance) < 1:
-                pass
-            else:
-                has_files, submitted_files, broken_instance = instance_has_files(namelist, instance)
+        for instances in plugin_instance_dict.values():
+            for instance in instances:
+                has_files, _, broken_instance = instance_has_files(namelist, [instance])
                 if not has_files:
-                    return False, f"\nYour {broken_instance} folder must contain an __init__.py and a test.py folder."
-                files.append(submitted_files)
+                    return False, f"\nYour {broken_instance} folder must contain all 4 required Python files:" \
+                                  f" __init__.py,  setup.py,  model.py and test.py."
```

Comment:
`model.py` is not required. Only two files are required: `__init__.py`, and `test.py`. `setup.py` is only necessary when requiring other dependencies, and `model.py` is a convention we're using to separate the code but not necessary (and also model-specific).